tests/src/python/test_provider_ogr.py

```diff
@@ -3497,6 +3498,41 @@ def testGeoJsonMapType(self):
         f = vl.getFeature(1)
         self.assertEqual(f.attributes()[0], {'style': {'color': 'yellow'}})
 
+    def testExtent(self):
+        # 2D points
+        datasource_2d = os.path.join(self.basetestpath, 'testExtent2D.csv')
+        with open(datasource_2d, 'w') as f:
+            f.write('id,WKT\n')
+            for i in range(9):
+                f.write(f'{i},POINT ({2*i} {i-3})\n')
+
+        vl = QgsVectorLayer(f'{datasource_2d}|layerid=0', 'test', 'ogr')
+        self.assertTrue(vl.isValid())
+        self.assertTrue(vl.featureCount(), 9)
+        self.assertEqual(vl.extent(), QgsRectangle(0, -3, 16, 5))
+        self.assertEqual(vl.extent3D(), QgsBox3D(0, -3, float('nan'), 16, 5, float('nan')))
+        del vl
+
+        os.unlink(datasource_2d)
+        self.assertFalse(os.path.exists(datasource_2d))
+
+        # 3D points
+        datasource_3d = os.path.join(self.basetestpath, 'testExtent3D.csv')
```

Comment:
you'll probably also want to do a similar test on GeoPackage to the the "SELECT min(ST_MinZ(...) ..." code path

Reply:
done!